ui/raidboss/emulator/data/Encounter.ts

```diff
@@ -68,46 +90,38 @@ export default class Encounter {
         }
       }
       if (res && res.groups && res.groups.language)
-        this.language = res.groups.language || this.language;
+        this.language = res.groups.language as Lang || this.language;
```

Comment:
```suggestion
        this.language = res.groups.language as Lang ?? this.language;
```

Maybe this is a case for a helper type guard function to turn string into Lang?

Reply:
Resolving this one in favor of the other comment thread.